### PR TITLE
Spooky 8 Ball

### LINKED ISSUE #12: [Feature Request] Spooky  8-ball
The goal of this task is to create an 8-ball that will generate random answers in a spookifying way, to support the halloween theme.

screen shot 2018-10-07 at 2 04 29 pm

A user will ask a the bot a question via a command `!8ball, is discord python awesome?` and the bot will answer just like an 8-ball. (hhmm)

screen shot 2018-10-07 at 2 02 13 pm

- [ ] list of spooky answers, 
- [ ] 8-ball image or emoji

### PR DESCRIPTION
**Pull request by RohanRadia**
Nice and clean eight ball setup. Made a few solid Halloween type responses. Made it so that either there will be a single message response or 2 messages to make it look like the eightball is thinking or talking to the 'powers above'.

Closes #12

### CODE REVIEW

File: bot/seasons/halloween/8ball.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+from discord.ext import commands
+
+log = logging.getLogger(__name__)
+
+with open(Path('bot', 'resources', 'halloween', 'responses.json'), 'r', encoding="utf8") as f:
+    responses = json.load(f)
+
+
+class EightBall(commands.Cog):
+    """Spooky Eightball answers."""
```
Comment (jodth07) [inline bot/seasons/halloween/8ball.py]: make the class/methods strings more meaningful. Such as, what it is, what it is doing, how it is doing it, as example.

File: bot/seasons/halloween/8ball.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            await ctx.send(choice[0])
+        else:
+            for x in choice:
+                await ctx.send(x)
+                asyncio.sleep(random.randint(2, 5))
+                await ctx.send(x)
+
+
+def setup(bot):
+    """Conversation starters Cog load."""
```
Comment (jodth07) [inline bot/seasons/halloween/8ball.py]: update this string

**Comment by jodth07** [comment]
closes #12

File: bot/seasons/halloween/8ball.py (hunk 2 above)
Comment (RohanRadia) [reply bot/seasons/halloween/8ball.py]: All the other cogs in easter have similar strings. Can you please be more precise? Not sure what to work on when told to 'update the string'. Anything specific?

File: bot/seasons/halloween/8ball.py (hunk 1 above)
Comment (RohanRadia) [reply bot/seasons/halloween/8ball.py]: Sorted.

**Comment by RohanRadia** [comment]
@jodth07 in terms of the sleep it should be an asyncio sleep.

**Comment by Suhail6inkling** [comment]
> @jodth07 in terms of the sleep it should be an asyncio sleep.

I think what he means by this is that if multiple people are using this command in the same channel, it isn't clear which output is for what question (if that makes sense)

**Comment by RohanRadia** [comment]
I feel you. We can sort out a little embed with the title as the question and the answer below.

File: bot/seasons/halloween/8ball.py (hunk 2 above)
Comment (sco1) [reply bot/seasons/halloween/8ball.py]: This is fine

**Comment by RohanRadia** [comment]
Regarding the sleep/embeds I haven't had a change to make the changes. Will attempt to get them made and sent today.

**Comment by jodth07** [comment]
> Regarding the sleep/embeds I haven't had a change to make the changes. Will attempt to get them made and sent today.

Whenever you are ready

**Comment by scOwez** [comment]
I finished implantation of a command, just need to add the results/answers of the 8ball

**Comment by sco1** [comment]
Hi @RohanRadia, have you found some time to make the desired updates?

**Comment by RohanRadia** [comment]
Hey @sco1 - Forgot about this. Will make the updates and submit within 72 hours. Sound good?

**Comment by RohanRadia** [comment]
Changes made @sco1 @jodth07 - ready for merge I believe. Made all changes as requested.

File: bot/seasons/halloween/8ball.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command(aliases=('spooky8ball',))
+    async def spookyeightball(self, ctx, *, question: str):
+        """Responds with a random response to a question."""
+        choice = random.choice(responses['responses'])
+        if len(choice) == 1:
+            await ctx.send(choice[0])
+        else:
```
Comment (sco1) [inline bot/seasons/halloween/8ball.py]: The difference in behavior between the single response sending a plantext message and the multi-response sending an embed is quite drastic. Let's make this consistent (and more compact) and just edit `msg.content` with the updated string.

File: bot/seasons/halloween/8ball.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import discord
+from discord.ext import commands
+
+log = logging.getLogger(__name__)
+
+with open(Path('bot', 'resources', 'halloween', 'responses.json'), 'r', encoding="utf8") as f:
+    responses = json.load(f)
+
+
+class EightBall(commands.Cog):
```
Comment (sco1) [inline bot/seasons/halloween/8ball.py]:
```suggestion
class SpookyEightBall(commands.Cog):
```

Minor tweak so there's less confusion between loading this vs. loading [evergreen's implementation](https://github.com/python-discord/seasonalbot/blob/master/bot/seasons/evergreen/magic_8ball.py).

File: bot/seasons/halloween/8ball.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+            msg = await ctx.send(embed=emb)
+            await asyncio.sleep(random.randint(2, 5))
+            emb.set_field_at(1, name='Answer', value=f'{choice[0]} \n {choice[1]}')
+            await msg.edit(embed=emb)
+
+
+def setup(bot):
+    """Spooky Eight Ball Cog Load."""
+
+    bot.add_cog(EightBall(bot))
```
Comment (sco1) [inline bot/seasons/halloween/8ball.py]:
```suggestion
    bot.add_cog(SpookyEightBall(bot))
```

As described above

File: bot/seasons/halloween/8ball.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    log.info("8Ball cog loaded")
```
Comment (sco1) [inline bot/seasons/halloween/8ball.py]:
```suggestion
    log.info("SpookyEightBall cog loaded")
```

As described above

**Comment by RohanRadia** [comment]
So you don't want an embed? @sco1 - I thought show in the questions and answer was decent. We could have an embed for the basic one too. Your choice. LMK.

**Comment by sco1** [comment]
We try to limit embeds as much as possible since they can take up a significant amount of room; I don't think this cog suffers by not having it and this keeps the behavior consistent with the existing 8ball command.

That aside, if we are including embeds then the behavior should be consistent across the cog.

**Pull request merged**